### ISSUE TITLE
export-name rules fails on all .tsx files

### ISSUE THREAD

**Issue by HamletDRC** [Microsoft/tslint-microsoft-contrib#121]
export-name rules fails on all .tsx files

**Comment by HamletDRC** [comment]
cannot reproduce now

**Comment by astorije** [comment]
Hey @HamletDRC, I'm wondering if I'm hitting this, and I can't explain what's going on.

In `test/helpers/enzyme-intl.tsx`, I have:

```tsx
import { mount } from 'enzyme';
import React, { ReactElement } from 'react';
import { IntlProvider } from 'react-intl';

export function mountWithIntl<P>(component: ReactElement<P>) {
  return mount(<IntlProvider locale="en">{component}</IntlProvider>);
}
```

This triggers:

```
.../test/helpers/enzyme-intl.tsx:5:1
ERROR: 5:1  export-name  The exported module or identifier name must match the file name. Found: .../test/helpers/enzyme-intl.tsx and mountWithIntl
```

However, I tried to build a dummy test case to identify the violation and potentially post a question somewhere, but couldn't reproduce it on a different example:

- `test/helpers/foobar-helpers.ts`:
  ```ts
  export function foo(): string {
    return 'foo';
  }
  ```
- `test/helpers/foobar-helpers.tsx`:
  ```tsx
  import React, { ReactNode } from 'react';

  const Foo = () => <span>Foo</span>;

  export function foo(): ReactNode {
    return <Foo />;
  }
  ```

Both of these pass the `export-name` rule!
Is there something obvious I am missing after a 12-hour-long day?!
